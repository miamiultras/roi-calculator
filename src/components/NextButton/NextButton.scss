.next-btn {
  a {
    text-decoration: none;
    &:visited {
      text-decoration: none;
    }
  }
  .container {
    display: flex;
    justify-content: center;
    button {
      width: 100%;
      display: flex;
      justify-content: space-around;
      align-items: center;
      font-family: 'Roboto';
      font-size: 16px;
      background: #5db98f;
      color: #fff;
      cursor: pointer;
      text-transform: uppercase;
      width: auto;
      padding: 10px;
      border: none;
    
      .square {
        margin-left: 10px;
        width: 24px;
        height: 24px;
        background-color: #fff;
        position: relative;
  
        i {
          border: solid #5db98f;
          border-width: 0 1.2px 1.2px 0;
          padding: 5px;
          position: absolute;
          right: 9px;
          top: 6.5px;
        } 
      
        .right {
            transform: rotate(-45deg);
            -webkit-transform: rotate(-45deg);
        }
      }
    }
  }
}