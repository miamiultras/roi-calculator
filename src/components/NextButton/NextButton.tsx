import React from 'react';
import { Link } from 'react-router-dom';

import './NextButton.scss';

const NextButton = ({ name, to, action }: { name: string, to: string, action?: any }) => (
  <div className='next-btn'>
    <Link to={`/calculator/${to}`}>
      <div className='container'>
        <button onClick={action}>
          <span>
            {name}
          </span>
          <div className='square'>
            <i className='arrow right' />
          </div>
        </button>
      </div>
    </Link>
  </div>
);

export default NextButton;
