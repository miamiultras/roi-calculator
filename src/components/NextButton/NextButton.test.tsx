import * as React from 'react';
import { shallow } from 'enzyme';
import NextButton from './NextButton';

function setup() {
  const props = {};
  const wrapper = shallow(<NextButton name='test' to='test' />);
  return {
    wrapper,
    props,
  };
}

describe('NextButton component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
