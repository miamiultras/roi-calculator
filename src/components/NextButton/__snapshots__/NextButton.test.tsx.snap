// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextButton component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NextButton
    name="test"
    to="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/calculator/test"
      >
        <div
          className="container"
        >
          <button>
            <span>
              test
            </span>
            <div
              className="square"
            >
              <i
                className="arrow right"
              />
            </div>
          </button>
        </div>
      </Link>,
      "className": "next-btn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
        >
          <button>
            <span>
              test
            </span>
            <div
              className="square"
            >
              <i
                className="arrow right"
              />
            </div>
          </button>
        </div>,
        "replace": false,
        "to": "/calculator/test",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button>
            <span>
              test
            </span>
            <div
              className="square"
            >
              <i
                className="arrow right"
              />
            </div>
          </button>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                test
              </span>,
              <div
                className="square"
              >
                <i
                  className="arrow right"
                />
              </div>,
            ],
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="arrow right"
                />,
                "className": "square",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow right",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/calculator/test"
        >
          <div
            className="container"
          >
            <button>
              <span>
                test
              </span>
              <div
                className="square"
              >
                <i
                  className="arrow right"
                />
              </div>
            </button>
          </div>
        </Link>,
        "className": "next-btn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="container"
          >
            <button>
              <span>
                test
              </span>
              <div
                className="square"
              >
                <i
                  className="arrow right"
                />
              </div>
            </button>
          </div>,
          "replace": false,
          "to": "/calculator/test",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              <span>
                test
              </span>
              <div
                className="square"
              >
                <i
                  className="arrow right"
                />
              </div>
            </button>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  test
                </span>,
                <div
                  className="square"
                >
                  <i
                    className="arrow right"
                  />
                </div>,
              ],
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="arrow right"
                  />,
                  "className": "square",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "arrow right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "div",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
